### DISCUSSION TITLE
HEIF/HEIC

### DISCUSSION THREAD

**Discussion by Petomai** [tannerhelland/PhotoDemon#396] [Q&A] [answered]
I recently upgraded my ancient phone to a slightly less ancient phone. It takes photos in HEIC, which it says saves space which is cool but literally nothing seems to be able to open this file format. I've tried Windows Photo Viewer and it says the file is corrupt, I tried Windows Photos it says I have to pay .99c to view my photos (lol) I tried VLC, Paint, and even PhotoDemon nothing can view these files except for my phone. Why is this? The phone I bought isn't very new, it's a Samsung Galaxy S10 Plus. Everywhere I read recommends converting the HEIC to Jpg with converters, but like that defeats the purpose of HEIC which is to save space. I might as well have a way to take photos in Jpg to begin with right? but as far as I can tell there is no way to do this. What's up with HEIF/HEIC being a pain like this?

**Comment by tannerhelland** [comment, marked as answer]
Hi @Petomai .  I share your frustration with HEIC support.

HEIC files are patent-encumbered, so developers must pay a fee to support it:

https://en.wikipedia.org/wiki/High_Efficiency_Image_File_Format#Patent_licensing

Apple and Samsung pay these fees on behalf of their users.  Microsoft does not.  This is why you must download the $0.99 codec from the Microsoft Store before accessing HEIC files in e.g. Windows Photo Viewer.

If you download that $0.99 codec, PhotoDemon, Paint.NET, and other Windows apps can "pick up" the ability to load HEIC files by passing them through the $0.99 codec.  It's a terrible workaround, but that's life as a Windows developer 🤷 

There are some open-source efforts to work around the patent issue.  They are currently complicated and challenging to work with, especially for a portable application like PhotoDemon.  I hope that these challenges will be mitigated with time, but at present, integrating them into PhotoDemon is not feasible.

In case this isn't confusing enough, there is a distinction between HEIF (which is a container format) and HEIC (which is a HEIF file containing HEVC data).  The HEIF container can use a number of different compression strategies, including non-patent-encumbered ones, which is why some HEIF files may be accessible on Windows via the free HEIF-specific download:

https://www.microsoft.com/en-us/p/heif-image-extensions/9pmmsr1cgpwg?activetab=pivot:overviewtab

That free download still does not help with HEIC images, however.
